overlord/snapstate/handlers_link_test.go

```diff
@@ -532,6 +532,50 @@ func (s *linkSnapSuite) TestDoUndoUnlinkCurrentSnapWithVitalityScore(c *C) {
 	c.Check(s.fakeBackend.ops, DeepEquals, expected)
 }
 
+func (s *linkSnapSuite) TestDoUnlinkSnapdUnlinks(c *C) {
```

Comment:
I'm surprised we don't need to modify more tests, I would have expected us to have a test that is essentially `TestDoLinkSnapdNthInstall` which we would see now has a different set of backend operations (where now that test would not unlink the old snap and would just link the new snap). Can we add such a test here to be more clear about the new behavior?

Reply:
I've added a snapstate level unit test.